src/hyperlink/_url.py

```diff
@@ -1991,9 +2005,20 @@ class DecodedURL(object):
         lazy (bool): Set to True to avoid pre-decode all parts of the URL to
             check for validity. Defaults to False.
 
+    .. note::
+
+      The :class:`DecodedURL` initializer takes a :class:`URL` object,
+      not URL components, like :class:`URL`. To programmatically
+      construct a :class:`DecodedURL`, you can use this pattern:
+
+        >>> print(DecodedURL().replace(scheme=u'https',
+        ... host=u'pypi.org', path=(u'projects', u'hyperlink')).to_text())
+        https://pypi.org/projects/hyperlink
+
+    .. versionadded:: 18.0.0
     """
 
-    def __init__(self, url, lazy=False):
+    def __init__(self, url=_EMPTY_URL, lazy=False):
```

Comment:
why the default?

Reply:
Explained in the other discussion, here: https://github.com/python-hyper/hyperlink/pull/126#discussion_r409186280